sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/protocol/blob_rest_client.hpp

```diff
@@ -3473,11 +3474,7 @@ namespace Azure { namespace Storage { namespace Blobs {
         std::set<std::string> metadataKeys;
         for (const auto& pair : options.Metadata)
         {
-          std::string key = pair.first;
-          std::transform(key.begin(), key.end(), key.begin(), [](unsigned char c) {
-            return static_cast<char>(std::tolower(c));
-          });
-          if (metadataKeys.insert(key).second == false)
+          if (metadataKeys.insert(Azure::Core::Details::ToLower(pair.first)).second == false)
```

Comment:
Should not be using Details namespace outside of the assembly.  We need to move our ToLower function if you are going to take a dependency on it.

Reply:
`ToLower` is a function provided in Core sdk. So this is a change core team should do, not storage team.

Reply:
Or are you suggesting that we should just keep things as it is. Should I just close this PR?

Reply:
FYI, @JeffreyRichter